app/src/androidTest/java/org/liberty/android/fantastischmemo/test/ui/FavouritesDatabaseTest.java

```diff
@@ -0,0 +1,97 @@
+package org.liberty.android.fantastischmemo.test.ui;
+
+import android.content.Intent;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+
+import junit.framework.Assert;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.liberty.android.fantastischmemo.common.AnyMemoDBOpenHelper;
+import org.liberty.android.fantastischmemo.common.AnyMemoDBOpenHelperManager;
+import org.liberty.android.fantastischmemo.entity.Card;
+import org.liberty.android.fantastischmemo.integrationtest.TestHelper;
+import org.liberty.android.fantastischmemo.test.AbstractExistingDBTest;
+import org.liberty.android.fantastischmemo.ui.StudyActivity;
+
+import java.io.File;
+import java.util.List;
+
+@RunWith(AndroidJUnit4.class)
+public class FavouritesDatabaseTest extends AbstractExistingDBTest {
+
+    private static final String SHUFFLE_CARDS = "shufflecards";
+    private Intent intent;
+    private String favouritesDBPath = "/sdcard/favourites.db";
+
+    @Rule
+    public ActivityTestRule<StudyActivity> studyActivityRule = new ActivityTestRule<StudyActivity>(StudyActivity.class, true, false);
+
+    @Before
+    public void setup(){
+        intent = new Intent();
+        intent.putExtra(StudyActivity.EXTRA_DBPATH, TestHelper.SAMPLE_DB_PATH);
+        intent.putExtra(SHUFFLE_CARDS, "false");
+        studyActivityRule.launchActivity(intent);
+        studyActivityRule.getActivity().emptyFavourtiesDeck();
+    }
+
+    @After
```

Comment:
Maybe put the after clause at the end of the file so it's more intuitive

Reply:
done